tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -30,14 +38,33 @@ static void init_test(void)
 	k_sem_reset(&rx_rdy);
 	k_sem_reset(&rx_buf_released);
 	k_sem_reset(&rx_disabled);
+#ifndef CONFIG_UART_WIDE_DATA
+	const struct uart_config uart_cfg = {
+		.baudrate = 115200,
+		.parity = UART_CFG_PARITY_NONE,
+		.stop_bits = UART_CFG_STOP_BITS_1,
+		.data_bits = UART_CFG_DATA_BITS_8,
+		.flow_ctrl = UART_CFG_FLOW_CTRL_NONE
+	};
+	__ASSERT_NO_MSG(uart_configure(uart_dev, &uart_cfg) == 0);
+#else
+	const struct uart_config uart_cfg = {
+		.baudrate = 115200,
+		.parity = UART_CFG_PARITY_NONE,
+		.stop_bits = UART_CFG_STOP_BITS_1,
+		.data_bits = UART_CFG_DATA_BITS_9,
+		.flow_ctrl = UART_CFG_FLOW_CTRL_NONE
+	};
+	__ASSERT_NO_MSG(uart_configure(uart_dev, &uart_cfg) == 0);
```

Comment:
Runtime config is optional and not supported by all async uart boards.

Reply:
True, but devicetree config isn't supported at all yet, so how would I get tests to execute on a uart that's configured for 9 bits data?

Reply:
Added an additional KConfig to enable runtime cfg